compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java

```diff
@@ -0,0 +1,2155 @@
+        }
+
+        BEnumSymbol enumSymbol = new BEnumSymbol(enumMembers, Flags.asMask(typeDefinition.flagSet),
+                names.fromIdNode(typeDefinition.name), names.fromIdNode(typeDefinition.name),
+                env.enclPkg.symbol.pkgID, definedType, env.scope.owner,
+                typeDefinition.pos, SOURCE);
+
+        enumSymbol.name = names.fromIdNode(typeDefinition.name);
+        enumSymbol.originalName = names.fromIdNode(typeDefinition.name);
+        enumSymbol.flags |= Flags.asMask(typeDefinition.flagSet);
+
+        enumSymbol.markdownDocumentation =
+                symEnter.getMarkdownDocAttachment(typeDefinition.markdownDocumentationAttachment);
+        enumSymbol.pkgID = env.enclPkg.packageID;
+        return enumSymbol;
+    }
+
+    public void resolveConstant(SymbolEnv symEnv, Map<String, BLangNode> modTable, BLangConstant constant) {
+        if (resolvingConstants.contains(constant)) { // To identify cycles.
+            dlog.error(constant.pos, DiagnosticErrorCode.CONSTANT_CYCLIC_REFERENCE,
+                    (this.resolvingConstants).stream().map(constNode -> constNode.symbol)
+                            .collect(Collectors.toList()));
+            constant.setBType(symTable.semanticError);
+            return;
+        }
+        resolvingConstants.add(constant);
+        if (resolvedConstants.contains(constant)) { // Already resolved constant.
+            resolvingConstants.remove(constant);
+            return;
+        }
+        constantTypeChecker.anonTypeNameSuffixes.push(constant.name.value);
+        defineConstant(symEnv, modTable, constant);
+        constantTypeChecker.anonTypeNameSuffixes.pop();
+        resolvingConstants.remove(constant);
+        resolvedConstants.add(constant);
+        checkUniqueness(constant);
+    }
+
+    private void checkUniqueness(BLangConstant constant) {
+        if (constant.symbol.kind == SymbolKind.CONSTANT) {
+            String nameString = constant.name.value;
+            BLangConstantValue value = constant.symbol.value;
+
+            if (constantMap.containsKey(nameString)) {
+                if (value == null) {
+                    dlog.error(constant.name.pos, DiagnosticErrorCode.ALREADY_INITIALIZED_SYMBOL, nameString);
+                } else {
+                    BLangConstantValue lastValue = constantMap.get(nameString);
+                    if (!value.equals(lastValue)) {
+                        if (lastValue == null) {
+                            dlog.error(constant.name.pos, DiagnosticErrorCode.ALREADY_INITIALIZED_SYMBOL, nameString);
+                        } else {
+                            dlog.error(constant.name.pos, DiagnosticErrorCode.ALREADY_INITIALIZED_SYMBOL_WITH_ANOTHER,
+                                    nameString, lastValue);
+                        }
+                    }
+                }
+            } else {
+                constantMap.put(nameString, value);
+            }
+        }
```

Comment:
Can we use early returns to get rid off the multiple levels of nested-ifs?
```suggestion
    if(constant.symbol.kind != SymbolKind.CONSTANT) {
        return;
    }
    
    String nameString = constant.name.value;
    BLangConstantValue value = constant.symbol.value;
    if(!constantMap.containsKey(nameString)) {
        constantMap.put(nameString, value);
        return;
    }
      
    if(value == null) {
        dlog.error(constant.name.pos, DiagnosticErrorCode.ALREADY_INITIALIZED_SYMBOL, nameString);
        return;
    }
        
    BLangConstantValue lastValue = constantMap.get(nameString);
    if (!value.equals(lastValue)) {
        if (lastValue == null) {
            dlog.error(constant.name.pos, DiagnosticErrorCode.ALREADY_INITIALIZED_SYMBOL, nameString);
            return;
        }
            
        dlog.error(constant.name.pos, DiagnosticErrorCode.ALREADY_INITIALIZED_SYMBOL_WITH_ANOTHER,
                                    nameString, lastValue);
   }
```